processors/base/helpers/__init__.py

```diff
@@ -241,33 +241,58 @@ def iter_rows(conn, dataset, table, orderby, bufsize=100, **filter):
             yield row
 
 
-def find_trial_by_identifiers(conn, identifiers, ignore_record_id=None):
-    """Find first trial matched by one of passed identifiers.
-
+def find_trial_by_identifiers(conn, identifiers, trial_public_title, source_id, record_id):
+    """Find if the trial already exists. Firstly, via identifiers. Secondly, via public title
```

Comment:
The method title, `find_trial_by_identifiers`, doesn't make sense anymore, as we're now also using `public_title`. In general, I feel that this method should be left mostly unmodified, and a new method created that uses it. Something like `find_trial(conn, trial)` that would encapsulate the logic of finding it via identifiers, `public_title` or whatever else we use in the future. This would also mean that the only place you'll modify, apart from this file, is `processors/base/writers/trial.py`.